packages/flutter/lib/src/widgets/form.dart

```diff
@@ -280,18 +280,21 @@ class FormState extends State<Form> {
         break;
     }
 
+    late Widget form;
     if (widget.canPop != null || (widget.onPopInvokedWithResult ?? widget.onPopInvoked) != null) {
-      return PopScope<Object?>(
+      form = PopScope<Object?>(
         canPop: widget.canPop ?? true,
         onPopInvokedWithResult: widget._callPopInvoked,
         child: _FormScope(formState: this, generation: _generation, child: widget.child),
       );
+    } else {
+      form = WillPopScope(
+        onWillPop: widget.onWillPop,
+        child: _FormScope(formState: this, generation: _generation, child: widget.child),
+      );
     }
 
-    return WillPopScope(
-      onWillPop: widget.onWillPop,
-      child: _FormScope(formState: this, generation: _generation, child: widget.child),
-    );
+    return Semantics(role: SemanticsRole.form, child: form);
```

Comment:
This may need container: true so that it won't merge up and accidentally get other children to be included in the form

Reply:
e.g something like this may break without the container:true here
```dart
Semantics(container:true, child: Column(
  children: [Semantics(container: true, child: Text('1')), Form(...)]

))
```
you will se that text('1') somehow become a child in the form in the semantics tree